src/dali_executor/dali_executor.h

```diff
@@ -42,9 +42,12 @@ struct shape_and_type_t {
 class DaliExecutor {
  public:
   DaliExecutor(
-      std::string serialized_pipeline, int max_batch_size, int device_id)
+      std::string serialized_pipeline, int max_batch_size, int num_threads, int device_id)
       : serialized_pipeline_(std::move(serialized_pipeline)),
-        device_id_(device_id), max_batch_size_(max_batch_size)
+        max_batch_size_(max_batch_size),
+        num_threads_(num_threads),
+        device_id_(device_id),
+        pipeline_(serialized_pipeline_, max_batch_size, num_threads, device_id)
```

Comment:
What if `serialized_pipeline_` if invalid and will throw?

Reply:
It's caught and handled when instantiating the model
https://github.com/triton-inference-server/dali_backend/blob/2b0f5139d72e093708eb42dffb0b7f940c7546ab/src/dali_backend.cc#L153

Reply:
OK